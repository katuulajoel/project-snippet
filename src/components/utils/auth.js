// eslint-disable-next-line no-unused-vars
import store from "../../store";

export function getAuth() {
  return store.getState().Auth;
}

export function getUser() {
  return getAuth().user;
}

export function isAdmin() {
  let user = getUser();
  return user.is_admin;
}

export function isPayAdmin() {
  let user = getUser();
  return user.is_pay_admin;
}

export function isDev() {
  return getUser().is_developer;
}

export function isDesigner() {
  return getUser().is_designer;
}

export function isClient() {
  return getUser().is_project_owner;
}

export function isPM() {
  return getUser().is_project_manager && !getUser().is_admin;
}

export function isAdminOrClient() {
  return isAdmin() || isClient();
}

export function isAdminOrPM() {
  return isAdmin() || isPM();
}

export function isPayAdminOrPM() {
  return isPayAdmin() || isPM();
}

export function isAdminOrPMOrClient() {
  return isAdmin() || isPM() || isClient();
}

export function isDevOrClient() {
  return isDev() || isClient();
}

export function isDevOrPM() {
  return isDev() || isPM();
}

export function getMyParticipation(project) {
  let myParticipation = null;
  if (project.participation) {
    (project.participation || []).forEach((item) => {
      if (item.user.id === getUser().id) {
        myParticipation = item;
      }
    });
  }
  return myParticipation;
}

export function isPendingProjectParticipant(project) {
  let isPending = false;
  if (isDev()) {
    let myParticipation = getMyParticipation(project);
    if (myParticipation) {
      return myParticipation.status === "initial";
    }
  }
  return isPending;
}

export function hasProjectAccess(project) {
  let allowedUserIds = [];
  ["user", "owner", "pm"].forEach((key) => {
    if (project[key]) {
      allowedUserIds.push(project[key].id);
    }
  });
  if (project.participation) {
    project.participation.forEach((item) => {
      if (item.status === "accepted" && item.user) {
        allowedUserIds.push(item.user.id);
      }
    });
  }
  return isAdmin() || allowedUserIds.includes(getUser().id);
}

export function isPMAndHasProjectAcess(project) {
  return hasProjectAccess(project) && isPM();
}

export function isAdminOrPMWithProjectAcess(project) {
  return isAdmin() || isPMAndHasProjectAcess(project);
}

export function isAdminOrClientOrPMWithProjectAcess(project) {
  return isAdmin() || isClient() || isPMAndHasProjectAcess(project);
}

export function isProjectClient(project) {
  const userId = getUser().id;
  if (project.owner && project.owner.id === userId) {
    return true;
  }

  if (project.user) {
    return project.user.id === userId && (!project.owner || !isAdmin());
  }
  return false;
}
