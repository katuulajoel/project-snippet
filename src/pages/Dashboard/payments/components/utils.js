import {
  isPayAdmin,
  isClient,
  isPayAdminOrPM,
} from "../../../../components/utils/auth";

export const GENERATE_INVOICE_ACTION = "GENERATE_INVOICE_ACTION";
export const PAY_ACTION = "PAY_ACTION";
export const EDIT_ACTION = "EDIT_ACTION";
export const DELETE_ACTION = "DELETE_ACTION";
export const MARK_AS_PAID_ACTION = "MARK_AS_PAID_ACTION";
export const ARCHIVE_ACTION = "ARCHIVE_ACTION";

export const showAction = (action, invoice) => {
  switch (action) {
    case GENERATE_INVOICE_ACTION:
      return (
        isPayAdmin() &&
        !invoice.finalized &&
        !invoice.last_sent_at &&
        !invoice.paid
      );
    case PAY_ACTION:
      return isClient() && invoice.finalized && !invoice.paid;
    case EDIT_ACTION:
    case DELETE_ACTION:
      return (
        isPayAdminOrPM() &&
        !invoice.paid &&
        !invoice.finalized &&
        !invoice.last_sent_at
      );
    case MARK_AS_PAID_ACTION:
      return (
        isPayAdmin() &&
        !invoice.paid &&
        (invoice.finalized || invoice.last_sent_at)
      );
    case ARCHIVE_ACTION:
      return isPayAdmin() && !invoice.paid;
    default:
      return false;
  }
};

// eslint-disable-next-line no-unused-vars
export const performAction = (action, invoice, props) => {
  switch (action) {
    case GENERATE_INVOICE_ACTION: {
      // onGenerateInvoice(invoice.id, props);
      break;
    }
    case PAY_ACTION: {
      /* openModal(<MakePaymentModal invoice={invoice} />, "Make Payment", true, {
        className: "modal-report modal-make-payment",
      }); */
      break;
    }
    case EDIT_ACTION: {
      // onUpdateInvoice(invoice, props);
      break;
    }
    case DELETE_ACTION: {
      // onDeleteInvoice(invoice.id, props);
      break;
    }
    case MARK_AS_PAID_ACTION: {
      // onMarkPaid(invoice.id, props);
      break;
    }
    case ARCHIVE_ACTION: {
      // onMarkArchived(invoice.id, props);
      break;
    }
  }
};
