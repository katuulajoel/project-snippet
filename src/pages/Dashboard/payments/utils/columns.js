// import { isPayAdmin } from "../../../../components/utils/auth";

//TODO: check if use is pay admin to active batch actions, add ...isPayAdmin() && on line 6

export const getTableColumns = (filter) => [
  ...(!(filter === "archived" || filter === "paid")
    ? [
        {
          Header: " ",
          accessor: "batch_action",
        },
      ]
    : []),
  {
    Header: "Date Created",
    accessor: "created_at",
  },
  {
    Header: "Client / Project / Payment Title",
    accessor: "title",
  },
  {
    Header: "Invoice No.",
    accessor: "invoice",
  },
  {
    Header: "Amount",
    accessor: "amount",
  },
  ...(filter !== "paid"
    ? [
        {
          Header: "Due Date",
          accessor: "due_date",
        },
        {
          Header: "Status",
          accessor: "status",
        },
        ...(filter !== "archived"
          ? [
              {
                Header: "",
                accessor: "actions",
              },
            ]
          : []),
      ]
    : []),
];
