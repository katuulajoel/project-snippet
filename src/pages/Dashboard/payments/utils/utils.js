import {
  isPayAdmin,
  isClient,
  isPayAdminOrPM,
} from "../../../../components/utils/auth";
import * as actions from "./constant"; //TODO: Load just actions needed
import {
  approvePayout,
  generateInvoice,
  markAsArchived,
  markAsPaid,
} from "./paymentActions";

function sumInvoiceAmounts(invoices) {
  return invoices
    .map((invoice) => {
      return invoice.amount || 0;
    })
    .reduce((total, number) => {
      return total + Math.round(number * 100) / 100;
    }, 0);
}

/**
 * Group invoices with the same reference into one batch object
 * @param {*} invoices list of invoices
 * @returns returns array of batch objects
 */
export function batchInvoices(invoices) {
  let batchedInvoices = {};
  invoices?.forEach((invoice) => {
    batchedInvoices[invoice.batch_ref] = [
      ...(batchedInvoices[invoice.batch_ref] || []),
      invoice,
    ];
  });

  let batchObjects = [];
  Object.keys(batchedInvoices).map((ref) => {
    let batchObj = {
      ...batchedInvoices[ref][0],
      invoices: batchedInvoices[ref],
    };
    batchObj.ref = ref;
    batchObj.amount = sumInvoiceAmounts(batchedInvoices[ref]);
    batchObjects.push(batchObj);
  });
  return batchObjects;
}

/**
 * Determine whether to show action or not depending on logged in User
 * @param {string} action constants used for the different actions in payments and payouts
 * @param {*} invoice invoice if payments action or invoices if payouts action
 * @returns {boolean} false | true
 */
export const showAction = (action, invoice) => {
  switch (action) {
    case actions.GENERATE_INVOICE_ACTION:
      return (
        isPayAdmin() &&
        !invoice.finalized &&
        !invoice.last_sent_at &&
        !invoice.paid
      );
    case actions.PAY_ACTION:
      return isClient() && invoice.finalized && !invoice.paid;
    case actions.EDIT_ACTION:
    case actions.DELETE_ACTION:
      return (
        isPayAdminOrPM() &&
        !invoice.paid &&
        !invoice.finalized &&
        !invoice.last_sent_at
      );
    case actions.MARK_AS_PAID_ACTION:
      return (
        isPayAdmin() &&
        !invoice.paid &&
        (invoice.finalized || invoice.last_sent_at)
      );
    case actions.APPROVE_BATCH_ACTION:
      return isPayAdmin() && invoice.status !== "approved";
    case actions.ARCHIVE_ACTION:
      return isPayAdmin() && !invoice.paid;
    case actions.EDIT_ACTION_BATCH:
    case actions.DELETE_ACTION_BATCH:
      return (
        isPayAdmin() &&
        !invoice.project.archived &&
        !invoice.status !== "approved" &&
        !invoice.paid
      );
  }
};

/**
 * Determine what function to call depending on the action that been triggered
 * @param {string} action constants used for the different actions in payments and payouts
 * @param {*} data invoice if payments action or invoices if payouts action
 */
export const performAction = (action, data) => {
  switch (action) {
    case actions.GENERATE_INVOICE_ACTION: {
      generateInvoice(data.id);
      break;
    }
    case actions.MARK_AS_PAID_ACTION: {
      markAsPaid(data.id);
      break;
    }
    case actions.ARCHIVE_ACTION: {
      markAsArchived(data.id);
      break;
    }
    case actions.APPROVE_BATCH_ACTION: {
      approvePayout(data);
      break;
    }
  }
};

/**
 * filters to be used as query params
 * @param {*} filter : selected payment status;
 * @returns object keys value pairs
 */
export const getPaymentsFilters = (filter) => {
  switch (filter) {
    case "paid":
      return { paid: "True" };
    case "pending":
      return { paid: "False", overdue: "False" };
    case "overdue":
      return { overdue: "True", paid: "False" };
    case "archived":
      return { archived: "True" };
    default:
      return {};
  }
};
