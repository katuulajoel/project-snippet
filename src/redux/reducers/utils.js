export const getIds = (items) => {
  return Array.from(
    new Set(
      items.map((item) => {
        return item.id;
      })
    )
  );
};

export const reduceUser = (state, user, profile, company) => {
  state = state || {};
  user = user || {};
  profile = { ...state.profile, ...user.profile, ...profile };
  company = { ...state.company, ...user.company, ...company };
  [
    ["name", "company"],
    ["bio", "company_bio"],
    ["website", "website"],
    ["vat_number", "vat_number"],
    ["reg_no", "company_reg_no"],
    ["ref_no", "reference_number"],
    ["skills", "skills"],
    ["country", "country"],
    ["city", "city"],
    ["street", "street"],
    ["plot_number", "plot_number"],
    ["postal_code", "postal_code"],
    ["postal_address", "postal_address"],
    ["tel_number", "phone_number"],
  ].forEach((item) => {
    let companyField = item[0],
      profileField = item[1];
    company[companyField] = company[companyField] || profile[profileField];
  });
  return { ...state, ...user, profile, company };
};
