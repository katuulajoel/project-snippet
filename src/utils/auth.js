// eslint-disable-next-line no-unused-vars
import store from "../redux/store";

export function getAuth() {
  return store.getState().Auth;
}

export function getUser() {
  return getAuth().user;
}

export function isAdmin() {
  let user = getUser();
  return user.is_admin;
}

export function isPayAdmin() {
  let user = getUser();
  return user.is_pay_admin;
}

export function isDev() {
  return getUser().is_developer;
}

export function isDesigner() {
  return getUser().is_designer;
}

export function isClient() {
  return getUser().is_project_owner;
}

export function isPM() {
  return getUser().is_client_manager && !getUser().is_admin;
}

export function isAdminOrClient() {
  return isAdmin() || isClient();
}

export function isAdminOrPM() {
  return isAdmin() || isPM();
}

export function isPayAdminOrPM() {
  return isPayAdmin() || isPM();
}

export function isAdminOrPMOrClient() {
  return isAdmin() || isPM() || isClient();
}

export function isDevOrClient() {
  return isDev() || isClient();
}

export function isDevOrPM() {
  return isDev() || isPM();
}

export function isPMAndHasProjectAcess(project) {
  return hasProjectAccess(project) && isPM();
}

export function isCM() {
  return getUser().is_client_manager && !getUser().is_admin;
}
export function isCSO() {
  return getUser().is_client_support_officer && !getUser().is_admin;
}

export function isCMOrCSOAndHasProjectAcess(project) {
  return hasProjectAccess(project) && (isCM() || isCSO());
}

export function isProjectClient(project) {
  const userId = getUser().id;
  if (project.owner && project.owner.id === userId) {
    return true;
  }

  if (project.user) {
    return project.user.id === userId && (!project.owner || !isAdmin());
  }
  return false;
}

export function hasProjectAccess(project) {
  let allowedUserIds = [];
  ["user", "owner", "cm"].forEach((key) => {
    if (project && project[key]) {
      allowedUserIds.push(project[key].id);
    }
  });
  if (project?.participation) {
    project.participation.forEach((item) => {
      if (item.status === "accepted" && item.user) {
        allowedUserIds.push(item.user.id);
      }
    });
  }
  return isAdmin() || allowedUserIds.includes(getUser().id);
}
