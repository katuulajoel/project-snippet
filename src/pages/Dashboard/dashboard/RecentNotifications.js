import React from "react";
import { useSelector, useDispatch } from "react-redux";
import { Link } from "react-router-dom";
import styled from "styled-components";
import {
  createNotificationLog,
  clearNotification,
} from "../../../redux/actions/DashboardActions";
import {
  STATUS_APPROVED,
  STATUS_PENDING,
} from "../../../configs/constants/projectConstants";
import Avatar from "../../../components/Avatar";
import IconButton from "../../../components/IconButton";
import SummaryPlaceholder from "../../../components/SummaryPlaceholder/SummaryPlaceholder";
import { getUser, isDev } from "../../../utils/auth";
import { getdateFormated } from "../../../utils/dateUtil";
import Progress from "../../../components/Progress";
import {
  invoice,
  participation,
  document,
  fieldChangeLog,
} from "./NotificationTypes";

const RecentNotifications = () => {
  const dispatch = useDispatch();
  const {
    notifications: { profile, activities },
    isMakingRequest: { notificationLog, clearNofitication, notification },
  } = useSelector(({ Dashboard }) => Dashboard);

  let shouldUpdateProfile =
      (profile.required.length > 0 || profile.optional.length > 0) &&
      !(profile.cleared || []).includes("complete"),
    shouldConnectPayoneer =
      isDev() &&
      ![STATUS_APPROVED, STATUS_PENDING].includes(getUser().payoneer_status) &&
      !(profile.cleared || []).includes("payoneer");

  const renderNotification = ({
    key,
    text,
    type,
    id,
    date,
    avatar = null,
    initials = null,
  } = {}) => {
    let classname = "";
    if (initials == "TA") {
      classname = "avatar-tunga";
    } else if (initials) {
      classname = "avatar-initials";
    }
    return (
      <div key={key} className="list-layout">
        <Avatar
          className={classname}
          image={avatar}
          size="dash"
          initials={initials}
        />
        <div className="detail">
          <p>{text}</p>
          <span className="sub-text">{getdateFormated(date, true)}</span>
        </div>
        <Close>
          <IconButton
            className="btn-icon"
            name="cross"
            size="sm"
            onClick={() =>
              dispatch(createNotificationLog({ type, notification_id: id }))
            }
          />
        </Close>
      </div>
    );
  };

  const notificationTypes = (item) => ({
    invoice: invoice(renderNotification, item),
    participation: participation(renderNotification, item),
    document: document(renderNotification, item),
    field_change_log: fieldChangeLog(renderNotification, item),
  });

  const clearAllNotifications = () => {
    dispatch(clearNotification());

    if (shouldUpdateProfile) {
      dispatch(
        createNotificationLog({ type: "profile", notification_id: "complete" })
      );
    }

    if (shouldConnectPayoneer) {
      dispatch(
        createNotificationLog({ type: "profile", notification_id: "payoneer" })
      );
    }
  };

  const notifications = activities
    .map((item, i) => {
      return notificationTypes({ ...item, key: `notification-${i}` })[
        item.activity_type
      ];
    })
    .filter((item) => item);

  return (
    <div className="dashboard-card">
      <h3 className="title">
        Recent Notifications
        {(notificationLog || clearNofitication || notification) && (
          <Progress style={{ display: "inline-block", marginLeft: "15px" }} />
        )}
        <small
          onClick={() => clearAllNotifications()}
          style={{
            float: "right",
            cursor: "pointer",
          }}
        >
          Clear all
        </small>
      </h3>

      {shouldUpdateProfile ||
      shouldConnectPayoneer ||
      notifications.length > 0 ? (
        <div className="content">
          {shouldUpdateProfile &&
            renderNotification({
              key: "profile-card",
              text: (
                <>
                  <span>
                    Please complete your profile{" "}
                    {profile.required.length > 0 && (
                      <span>
                        to be able to {isDev() ? "accept" : "create"} projects
                      </span>
                    )}
                  </span>
                  &nbsp;
                  <Link to={`/settings/`}>Go to profile</Link>
                </>
              ),
              type: "profile",
              id: "complete",
              initials: "TA",
            })}

          {shouldConnectPayoneer &&
            renderNotification({
              key: "payoneer-card",
              text: (
                <>
                  Please connect your account with Payoneer to receive
                  payments&nbsp;
                  <Link to={`/settings/payment/`}>Set up Payoneer</Link>
                </>
              ),
              type: "profile",
              id: "payoneer",
              initials: "TA",
            })}

          {notifications}
        </div>
      ) : (
        <SummaryPlaceholder description="No recent notifications" />
      )}
    </div>
  );
};

const Close = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  margin: 0 0 0 auto;

  .btn-icon {
    color: #8f9bb3;
    i {
      font-weight: 600;
    }
  }
`;

export default RecentNotifications;
