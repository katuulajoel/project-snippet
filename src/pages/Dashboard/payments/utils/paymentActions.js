import PropTypes from "prop-types";
import React from "react";
import moment from "moment";
import {
  archiveInvoice,
  generateInvoice as generateInvoiceAction,
  updateInvoice,
  downloadInoicesCsv,
  getInvoiceSummary,
  bulkAction,
} from "../../../../actions/InvoiceActions";
import DateRangeForm from "../../../../components/DateRangeForm";
import ModalHeader from "../../../../components/ModalHeader";
import { openConfirm, openModal } from "../../../../components/utils/modals";
import store from "../../../../store";

const Header = (props) => (
  <ModalHeader
    {...props}
    style={{ paddingBottom: "8px" }}
    options={props.title}
  />
);

Header.propTypes = {
  title: PropTypes.string,
};

export function generateInvoice(invoiceId) {
  openConfirm(
    <div>Are you sure you want to generate an invoice for this payment?</div>,
    null,
    true,
    { ok: "Yes" },
    <Header title="Generate Invoice" />
  ).then(() => {
    store.dispatch(generateInvoiceAction(invoiceId));
  });
}

export function markAsPaid(invoiceId) {
  openConfirm(
    <div>Are you sure you want to mark this invoice as paid?</div>,
    null,
    true,
    { ok: "Yes" },
    <Header title="Mark as Paid" />
  ).then(() => {
    store.dispatch(updateInvoice(invoiceId, { paid: true }));
  });
}

export function markAsArchived(invoiceId) {
  openConfirm(
    <div>Are you sure you want to archive this invoice?</div>,
    null,
    true,
    { ok: "Yes" },
    <Header title="Archive invoice" />
  ).then(() => {
    store.dispatch(archiveInvoice(invoiceId));
  });
}

export function approvePayout(invoices) {
  openConfirm(
    <div>Are you sure you want to approve this payout?</div>,
    null,
    true,
    { ok: "Approve", cancel: "Cancel" },
    <Header title="Approve payout" />
  ).then(() => {
    invoices.forEach((invoice) => {
      store.dispatch(updateInvoice(invoice.id, { status: "approved" }));
    });
  });
}

export function downloadCsv(filter, type) {
  openModal(
    <DateRangeForm id={`export-form`} />,
    "",
    true,
    {
      className: "modal-payments",
      ok: `Download CSV`,
      form: {
        type: "submit",
        form: `export-form`,
      },
    },
    <Header title={`Export ${type}`} />,
    false
  ).then((data) => {
    if (data) {
      let params = {};

      switch (filter) {
        case "archived":
          params["archived"] = "True";
          params["paid"] = "False";
          break;
        case "all":
          params["archived"] = "False";
          break;
        case "pending":
          params["archived"] = "False";
          params["paid"] = "False";
          params["overdue"] = "False";
          break;
        case "overdue":
          params["archived"] = "False";
          params["paid"] = "False";
          params["overdue"] = "True";
          break;
        case "paid":
          params["archived"] = "False";
          params["paid"] = "True";
      }

      store.dispatch(
        downloadInoicesCsv({
          start: data.start,
          end: data.end,
          type: type === "Payments" ? "sale" : "purchase",
          ...params,
        })
      );
    }
  });
}

export function filterPayment({ start, end }, type, setSummariesRange) {
  openModal(
    <DateRangeForm
      id={`fiter-payment`}
      defaultStart={start}
      defaultEnd={end}
      maxdate={new Date()}
      message={"Select date range for filter"}
    />,
    "",
    true,
    {
      className: "modal-payments",
      ok: `Submit`,
      form: {
        type: "submit",
        form: `fiter-payment`,
      },
    },
    <ModalHeader
      style={{ paddingBottom: "8px" }}
      options={{
        title: `Filter Total ${type === "Payments" ? "Payments" : "Payouts"}`,
      }}
    />,
    false
  ).then((data) => {
    if (data) {
      const START = `${moment(data.start).format(
        moment.HTML5_FMT.DATE
      )}T00:00:00`;
      const END = `${moment(data.end).format(moment.HTML5_FMT.DATE)}T23:59:59`;

      setSummariesRange({
        start: START,
        end: END,
      });

      store.dispatch(
        getInvoiceSummary({
          min_date: START,
          max_date: END,
          type: type === "Payments" ? "sale" : "purchase",
        })
      );
    }
  });
}

export function bulkMarkAsPaid(invoices) {
  openConfirm(
    <div>Are you sure you want to mark all selected as paid?</div>,
    null,
    true,
    { ok: "Yes", cancel: "Cancel" },
    <Header title="Bulk mark as Paid" />
  ).then(() => {
    invoices.forEach((invoice) => {
      store.dispatch(updateInvoice(invoice.id, { paid: true }));
    });
  });
}

export function bulkGenerateInvoice(invoices) {
  openConfirm(
    <div>Are you sure you want to generate invoices for all selected?</div>,
    null,
    true,
    { ok: "Yes", cancel: "Cancel" },
    <Header title="Bulk generate invoices" />
  ).then(() => {
    invoices.forEach((invoice) => {
      store.dispatch(generateInvoice(invoice.id, { paid: true }));
    });
  });
}

export function bulkDeleteInvoice(invoices) {
  openConfirm(
    <div>Are you sure you want to delete all selected?</div>,
    null,
    true,
    { ok: "Yes", cancel: "Cancel" },
    <Header title="Bulk delete invoices" />
  ).then(() => {
    store.dispatch(bulkAction(invoices, "archive"));
  });
}

export function bulkArchiveInvoice(invoices) {
  openConfirm(
    <div>Are you sure you want to archive all selected?</div>,
    null,
    true,
    { ok: "Yes", cancel: "Cancel" },
    <Header title="Bulk archive invoices" />
  ).then(() => {
    store.dispatch(bulkAction(invoices, "delete"));
  });
}
