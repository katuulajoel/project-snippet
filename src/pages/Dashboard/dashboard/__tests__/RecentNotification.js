/* eslint-disable no-unused-vars */
import React from "react";
import renderer from "react-test-renderer";
import RecentNotifications from "../RecentNotifications";
import { BrowserRouter } from "react-router-dom";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";
import * as actions from "../../../../utils/auth";
import { mount } from "enzyme/build";

const middlewares = [thunk];

const mockAppStore = (state) => {
  const mockStore = configureStore(middlewares);
  return mockStore(state);
};

const state = {
  Dashboard: {
    notifications: {
      profile: {
        required: [{}],
        optional: [],
        cleared: ['complete', 'payoneer'],
      },
      activities: [
        {
          action: 'create',
          activity_type: 'participation',
          description: null,
          id: 573,
          activity: {
            created_by: {
              display_name: 'Admin Tunga',
            },
          },
        },
        {
          action: 'create',
          activity_type: 'document',
          description: null,
          id: 534,
          activity: {
            created_by: {
              display_name: 'Admin Tunga',
            },
          },
        },
      ],
    },
    isMakingRequest: {},
  },
  Invoice: {
    search: {},
  },
};

describe('Dashboard test', () => {
  beforeEach(() => {
    jest.spyOn(actions, 'isDev').mockReturnValueOnce(true);
    jest.spyOn(actions, 'getUser').mockReturnValueOnce({ payoneer_status: 'approved' });
  });

  /* it("Should match snapshot test", () => {
    const mockDate = new Date(1466424490000);
    const spy = jest.spyOn(global, "Date").mockImplementation(() => mockDate);

    const tree = renderer
      .create(
        <BrowserRouter>
          <Provider store={mockAppStore(state)}>
            <ThemeProvider theme={theme}>
              <RecentNotifications />
            </ThemeProvider>
          </Provider>
        </BrowserRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
    spy.mockRestore();
  });
 */
  it('should clear all notifications', () => {
    const store = mockAppStore(state);
    const wrapper = mount(
      <BrowserRouter>
        <Provider store={store}>
          <RecentNotifications />
        </Provider>
      </BrowserRouter>
    );

    var clearAllBtn = wrapper.find('.title small');
    clearAllBtn.simulate('click');

    const expectedActions = [
      {
        type: 'DELETE_NOTIFICATIONS_START',
      },
    ];

    expect(store.getActions()).toEqual(expectedActions);
  });

  it('should clear notification', () => {
    const store = mockAppStore(state);
    const wrapper = mount(
      <BrowserRouter>
        <Provider store={store}>
          <RecentNotifications />
        </Provider>
      </BrowserRouter>
    );

    var clearBtn = wrapper.find('.list-layout .btn-icon').at(0);
    clearBtn.simulate('click');

    const expectedActions = [
      {
        type: 'CREATE_NOTIFICATION_LOG_START',
      },
    ];

    expect(store.getActions()).toEqual(expectedActions);
  });
});
