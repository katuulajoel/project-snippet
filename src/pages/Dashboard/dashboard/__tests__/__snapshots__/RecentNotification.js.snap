// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard test Should match snapshot test 1`] = `
<div
  className="dashboard-card"
>
  <h3
    className="title"
  >
    Recent Notifications
    <small
      onClick={[Function]}
      style={
        Object {
          "cursor": "pointer",
          "float": "right",
        }
      }
    >
      Clear all
    </small>
  </h3>
  <div
    className="content"
  >
    <div
      className="list-layout"
    >
      <div
        className="sc-bdnxRM eNsiXn"
      >
        <div
          className="avatar "
          style={Object {}}
        >
          <i
            className="tg-ic-avatar  "
          />
        </div>
      </div>
      <div
        className="detail"
      >
        <p>
          <span>
            You have
             
            been added to 
            your
             team forÂ 
            <a
              href="/projects/undefined/team"
              onClick={[Function]}
            />
          </span>
        </p>
        <span
          className="sub-text"
        >
          Today, 03:08 PM
        </span>
      </div>
      <div
        className="sc-dlnjwi iqEnIB"
      >
        <button
          className="btn btn-icon sc-gtsrHT fvXHXw btn-icon  "
          form={null}
          onClick={[Function]}
          type="button"
        >
          <i
            className="tg-ic-cross tunga-ic-sz-sm "
          />
        </button>
      </div>
    </div>
    <div
      className="list-layout"
    >
      <div
        className="sc-bdnxRM eNsiXn avatar-initials"
      >
        <div
          className="avatar avatar-initials"
          style={Object {}}
        >
          AT
        </div>
      </div>
      <div
        className="detail"
      >
        <p>
          <span>
            You
             added a undefined document to project 
            <a
              href="/projects/undefined/docs"
              onClick={[Function]}
            />
          </span>
        </p>
        <span
          className="sub-text"
        >
          Today, 03:08 PM
        </span>
      </div>
      <div
        className="sc-dlnjwi iqEnIB"
      >
        <button
          className="btn btn-icon sc-gtsrHT fvXHXw btn-icon  "
          form={null}
          onClick={[Function]}
          type="button"
        >
          <i
            className="tg-ic-cross tunga-ic-sz-sm "
          />
        </button>
      </div>
    </div>
  </div>
</div>
`;
